activerecord/lib/active_record/relation/predicate_builder/array_handler.rb

```diff
@@ -9,6 +9,16 @@ def call(attribute, value)
         values = value.map { |x| x.is_a?(Base) ? x.id : x }
         nils, values = values.partition(&:nil?)
```

Comment:
Can drop the nil-partitioning here, too, and set a nil-present flag in the `value.map` to avoid having to loop through the values again.